modules/websocket_bridge/include/hephaestus/utils/ws_client.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <chrono>
+#include <csignal>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <foxglove/websocket/websocket_client.hpp>
+#include <foxglove/websocket/websocket_notls.hpp>
+#include <google/protobuf/message.h>
+#include <hephaestus/utils/ws_protocol.h>
+
+namespace heph::ws {
+
+using WsClientNoTls = foxglove::Client<foxglove::WebSocketNoTls>;
+
+struct ServiceCallState {
+  enum class Status : std::uint8_t { SUCCESS = 0, DISPATCHED = 1, FAILED = 2 };
+
+  uint32_t call_id;
+  Status status;
+  std::chrono::steady_clock::time_point dispatch_time;
+  std::chrono::steady_clock::time_point response_time;
+
+  std::optional<WsServiceResponse> response;
+  std::string error_message;
+
+  explicit ServiceCallState(uint32_t call_id);
+
+  auto receiveResponse(const WsServiceResponse& service_response, WsAdvertisements& ws_server_ads)
+      -> std::optional<std::unique_ptr<google::protobuf::Message>>;
```

Comment:
the whole serdes module is supposed to be agnostic from the underlying serialization mechanism, so here we could return just the byte buffer.
Or we want to explicitly state that this bridge works only for protobuf messages?

Reply:
this is a convenience function used in the ws client examples. I could do it in two functions, one returning bytes and the other then the proto message, but the only other encoding option to support that even makes sense for this bridge is json, so I don't think this makes sense and would probably have a lot more places to rework.

Reply:
ok, let's keep it as it is. I think we can safely assume for now we support only proto and then extend it later if needed